src/main/java/kg/jarkyn/cobspecserver/Server.java

```diff
@@ -4,11 +4,15 @@
 import kg.jarkyn.cobspecserver.sockets.ClientSocket;
 import kg.jarkyn.cobspecserver.sockets.ListenerSocket;
 
+import java.util.concurrent.ExecutorService;
+
 public class Server {
+    private ExecutorService executor;
     private ListenerSocket listener;
     private ResponseController responseController;
 
-    public Server(ListenerSocket listener, ResponseController responseController) {
+    public Server(ExecutorService executor, ListenerSocket listener, ResponseController responseController) {
```

Comment:
Creating dependencies to concrete classes is a typical smell. There is a missing opportunity to unveil an interesting abstraction.

Why to put the multithreading code here? it seems like `responseController` knows how to respond to a `client`. It could delegate to workers as well.

But if we do that, we might put too much responsibilities on that concrete class. This might give you a hint that an abstraction is wanting to appear.
That could guide you to the composite pattern: https://en.wikipedia.org/wiki/Composite_pattern